### PR TITLE
fix(trace): update proofList struct

### PR DESCRIPTION
**Pull request by 0xmountaintop**

### CODE REVIEW

**Comment by ChuhanJin** [comment]
Can one of the admins verify this patch?

File: core/vm/logger_trace.go [hunk 1]
```diff
@@ -128,13 +113,52 @@ func traceLastNAddressProof(n int) traceFunc {
...
-		extraData.ProofList = append(extraData.ProofList, encodeProof(proof))
+		extraData.ProofList = append(extraData.ProofList, proof)
 	}
 	return err
 }
 
+func getWrappedProofForAddr(l *StructLogger, address common.Address) (*types.AccountProofWrapper, error) {
+	proof, err := l.env.StateDB.GetProof(address)
+	if err != nil {
+		return &types.AccountProofWrapper{}, err
```
Comment (mask-pp) [inline core/vm/logger_trace.go]: Use `return nil, err` is much better?

File: core/vm/logger_trace.go [hunk 2]
```diff
@@ -128,13 +113,52 @@ func traceLastNAddressProof(n int) traceFunc {
...
+		Balance: l.env.StateDB.GetBalance(address),
+		Proof:   encodeProof(proof),
+		Storage: &types.StorageProofWrapper{},
+	}, nil
+}
+
+func getWrappedProofForStorage(l *StructLogger, address common.Address, key common.Hash) (*types.AccountProofWrapper, error) {
+	proof, err := l.env.StateDB.GetProof(address)
+	if err != nil {
+		return &types.AccountProofWrapper{}, err
```
Comment (mask-pp) [inline core/vm/logger_trace.go]: Same as above

File: core/vm/logger_trace.go [hunk 3]
```diff
@@ -128,13 +113,52 @@ func traceLastNAddressProof(n int) traceFunc {
...
+	}
+
+	storageProof, err := l.env.StateDB.GetStorageProof(address, key)
+	if err != nil {
+		return &types.AccountProofWrapper{}, err
```
Comment (mask-pp) [inline core/vm/logger_trace.go]: Same as above

File: core/vm/logger_trace.go [hunk 4]
```diff
@@ -128,13 +113,52 @@ func traceLastNAddressProof(n int) traceFunc {
...
+	}
+
+	return &types.AccountProofWrapper{
+		Address: address,
+		Nonce:   l.env.StateDB.GetNonce(address),
+		Balance: l.env.StateDB.GetBalance(address),
+		Proof:   encodeProof(proof),
+		Storage: &types.StorageProofWrapper{},
```
Comment (mask-pp) [inline core/vm/logger_trace.go]: It's needless if `Storage` is empty?

**Review by Lawliet-Chan — APPROVED** [review]

**Review by mask-pp — APPROVED** [review]

**Pull request merged**